### PR TITLE
Scripts: Remove broken `lint-md-js` command

### PR DESCRIPTION
**Pull request by gziolo**
## What?

I tested `npm run lint:md-js` in the Gutenberg repository and directly with `wp-scripts lint-md-jd` and they both are completely broken. 

## Why?

It looks like we use too many custom ESLint rules in the recommended preset that error when used in the context of the Markdown files. I tried disabling all of them (related to i18n and Prettier) but it still produces hundreds of errors because we use too many incomplete code examples. I don't think it's realistic to expect that we can successfully lint JavaScript snippets with the quite strict set of rules we maintain.

## How?

I removed the `lint-md-js` command and all related documentation.

### CODE REVIEW

File: packages/scripts/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
 
 ## Unreleased
 
+## Breaking Changes
+
+-   Remove `lint-md-js` script that was broken for some time and it's extemely hard to make it work correctly with the recommended ESLint config in Markdown files.
+
+## 22.5.0 (2022-04-21)
```
Comment (gziolo) [inline packages/scripts/CHANGELOG.md:9]: This one was misplaced 🙈

File: packages/scripts/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
+
+### Bug Fixes
+
+-   Update `markdownlint` and `markdownlint-cli` to the latest versions ([#40479](https://github.com/WordPress/gutenberg/pull/40479)).
```
Comment (gziolo) [inline packages/scripts/CHANGELOG.md:17]: This one was misplaced 🙈

File: packages/scripts/README.md [hunk 3]
```diff
@@ -48,8 +47,8 @@ _Example:_
...
 
 ## Automatic block.json detection and the source code directory
-When using the `start` or `build` commands, the source code directory ( the default is `./src`) and its subdirectories are scanned for the existence of `block.json` files. If one or more are found, they are treated a entry points and will be output into corresponding folders in the `build` directory. The allows for the creation of multiple blocks that use a single build process. The source directory can be customized using the `--webpack-src-dir` flag.
 
+When using the `start` or `build` commands, the source code directory ( the default is `./src`) and its subdirectories are scanned for the existence of `block.json` files. If one or more are found, they are treated a entry points and will be output into corresponding folders in the `build` directory. The allows for the creation of multiple blocks that use a single build process. The source directory can be customized using the `--webpack-src-dir` flag.
```
Comment (ryanwelcher) [inline packages/scripts/README.md]:
```suggestion
When using the `start` or `build` commands, the source code directory ( the default is `./src`) and its subdirectories are scanned for the existence of `block.json` files. If one or more are found, they are treated a entry points and will be output into corresponding folders in the `build` directory. This allows for the creation of multiple blocks that use a single build process. The source directory can be customized using the `--webpack-src-dir` flag.
```

Comment (ryanwelcher) [reply packages/scripts/README.md]: Small typo ( of my own doing I believe 😄  )

**Review by ryanwelcher — APPROVED** [review]

**Pull request merged**